components/api_version/src/api_v1.rs

```diff
@@ -43,28 +43,18 @@ impl KvFormat for ApiV1 {
     ) -> Result<Key> {
         match src_api {
             ApiVersion::V1 | ApiVersion::V1ttl => Ok(Key::from_encoded_slice(key)),
-            ApiVersion::V2 => {
-                debug_assert_eq!(ApiV2::parse_key_mode(key), KeyMode::Raw);
-                let (mut user_key, _) = ApiV2::decode_raw_key(&Key::from_encoded_slice(key), true)?;
-                user_key.remove(0); // remove first byte `RAW_KEY_PREFIX`
-                Ok(Self::encode_raw_key_owned(user_key, None))
-            }
+            ApiVersion::V2 => unreachable!(), // reject apiv2 -> apiv1 conversion
```

Comment:
It will panic TiKV if the arm matches. What about returning an error?

Reply:
backup process do check the apiversion in https://github.com/tikv/tikv/blob/ba391ff506c8b7f4b0cd7c9ef0b9f04ce87c3d7e/components/backup/src/utils.rs#L127-L133 to ensure that this branch is unreachable.  Handle error will make backup process a little complicated here https://github.com/tikv/tikv/blob/ba391ff506c8b7f4b0cd7c9ef0b9f04ce87c3d7e/components/backup/src/endpoint.rs#L213-L216. And we want take this pr in v6.1.0 release, so we need make min changes.

Reply:
Agree should return error instead of just panicking.

Reply:
@overvenus @BusyJay error handling is added, PTAL~ thanks.